### PR TITLE
chore: add basic User model tests for RiffTube backend

### LINKED ISSUE #171: Set up Basic Model Tests for RiffTube Backend
# 🧪 Ticket: Set up Basic Model Tests for RiffTube Backend

## Overview

We need to establish basic backend tests for our RiffTube models using Minitest.  
This ticket covers setting up model tests for the **core models**:  
- `User`  
- `Project`  
- `Riff`  
- `RiffVersion`  
- `ProjectRiff`  
- `MediaFile`  
- `Comment`

The goal is to validate basic functionality (presence validations, associations if appropriate) and ensure the foundation is solid for future feature testing.

## Tasks

- [ ] Create `test/models/` test files for the following models:
  - [ ] `user_test.rb`
  - [ ] `project_test.rb`
  - [ ] `riff_test.rb`
  - [ ] `riff_version_test.rb`
  - [ ] `project_riff_test.rb`
  - [ ] `media_file_test.rb`
  - [ ] `comment_test.rb`
- [ ] For each model, write **at least two** tests:
  - **Presence Test**: Model cannot save without required fields (e.g., no email for User, no title for Project)
  - **Valid Object Test**: Model saves successfully when required fields are provided
- [ ] Run `rails test` and confirm all tests pass
- [ ] Fix any failures if necessary (adjust validations if missing)

---

## Example Test Case (User)

```ruby
require "test_helper"

class UserTest < ActiveSupport::TestCase
  test "should not save user without email" do
    user = User.new
    assert_not user.save, "Saved the user without an email"
  end

  test "should save valid user" do
    user = User.new(email: "test@example.com")
    assert user.save, "Could not save a valid user"
  end
end
```

---

## Example Test Case (Project)

```ruby
require "test_helper"

class ProjectTest < ActiveSupport::TestCase
  test "should not save project without title" do
    project = Project.new
    assert_not project.save, "Saved the project without a title"
  end

  test "should save valid project" do
    project = Project.new(title: "My Awesome Project", owner_id: some_user.id, video_host: "youtube")
    assert project.save, "Could not save a valid project"
  end
end
```

---

## Definition of Done ✅

- [ ] There is a test file for each major model under `test/models/`
- [ ] Each model test contains at least:
  - [ ] One **invalid** save test (missing required fields)
  - [ ] One **valid** save test (with all required fields)
- [ ] `rails test` runs cleanly with no failures
- [ ] Tests use plain, clear Minitest style (`assert`, `assert_not`)
- [ ] Tests help verify that basic model validations are working
- [ ] No regressions or broken app behavior after adding tests

---

# 🧹 Bonus (Optional)

- [ ] Consider adding more tests for edge cases (e.g., invalid formats, enum checks) if time allows

---

### PR DESCRIPTION
**Pull request by JossWritesCode**
Resolves #171
image

### CODE REVIEW

**Review by paxfeline — APPROVED** [review]
Great work. I'm going to study the factory stuff and model tests.

File: backend/app/models/user.rb [hunk 1]
```diff
@@ -41,9 +46,13 @@ class User < ApplicationRecord
 
   private
 
-  def password_required_unless_oauth
-    if provider.blank? && password_digest.blank?
-      errors.add(:password, "can't be blank")
+  def password_required?
+    provider.blank? || (!password_digest.present? && provider_changed?)
```
[the remark this answers is not in the document]
Comment (JossWritesCode) [reply backend/app/models/user.rb:50]: No, I want to allow valid OAuth users to exist without passwords. I only want to require a password if it's needed.
